components/notebook-controller/pkg/culler/culler.go

```diff
@@ -0,0 +1,194 @@
+package culler
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("culler")
+var client = &http.Client{
+	Timeout: time.Second * 10,
+}
+
+// The constants with name 'DEFAULT_{ENV_Var}' are the default values to be
+// used, if the respective ENV vars are not present.
+// All the time numbers correspond to minutes.
+const DEFAULT_IDLE_TIME = "1440" // One day
+const DEFAULT_CULLING_CHECK_PERIOD = "1"
+const DEFAULT_ENABLE_CULLING = "false"
+const STOP_ANNOTATION = "kubeflow-resource-stopped"
```

Comment:
Could you add some comment for the annotation (what does it mean, when will we add it)

Reply:
Done, might have overdone it a little bit with the verbosity but the description should be clear

Reply:
I am a little confused. If the controller is setting the annotation, and is also checking the annotation, why do we need the annotation? Doesn't controller just know whether it should be culled?

Reply:
My apologies, I didn't provide enough details regarding that.
Yes, the controller knows from the start if the Notebook should be culled and could do it without the annotation. 

I decided to set an annotation to make it easier for the UI later to check that a Notebook is (being) stopped. Also, if the UI would want to start/stop a Notebook then it would only need to modify the annotation on the CR.

Reply:
I see